src/chrome/breakOnLoadHelper.ts

```diff
@@ -157,14 +163,33 @@ export class BreakOnLoadHelper {
         const hitBreakpoints = notification.hitBreakpoints;
         let allStopOnEntryBreakpoints = true;
 
+        const pausedScriptId = notification.callFrames[0].location.scriptId;
+        const pausedScriptUrl = this._chromeDebugAdapter.scriptsById.get(pausedScriptId).url;
+        const mappedUrl = await this._chromeDebugAdapter.pathTransformer.scriptParsed(pausedScriptUrl);
+
         // If there is a breakpoint which is not a stopOnEntry breakpoint, we appear as if we hit that one
         // This is particularly done for cases when we end up with a user breakpoint and a stopOnEntry breakpoint on the same line
-        hitBreakpoints.forEach(bp => {
-            if (!this._stopOnEntryBreakpointIdToRequestedFileName.has(bp)) {
+        for (let bp of hitBreakpoints) {
+            let regexAndFileNames = this._stopOnEntryBreakpointIdToRequestedFileName.get(bp);
+            if (!regexAndFileNames) {
                 notification.hitBreakpoints = [bp];
                 allStopOnEntryBreakpoints = false;
+            } else {
+                const normalizedMappedUrl = this._chromeDebugAdapter.fixPathCasing(mappedUrl);
+                if (regexAndFileNames.fileSet.has(normalizedMappedUrl)) {
+                    regexAndFileNames.fileSet.delete(normalizedMappedUrl);
+                    assert(this._stopOnEntryRequestedFileNameToBreakpointId.delete(normalizedMappedUrl), `Expected to delete break-on-load information associated with url: ${normalizedMappedUrl}`);
+
+                    if (regexAndFileNames.fileSet.size === 0) {
+                        logger.log(`Stop on entry breakpoint hit for last remaining file. Removing: ${bp} created for: ${normalizedMappedUrl}`);
+                        await this.removeBreakpointById(bp);
+                        assert(this._stopOnEntryRegexToBreakpointId.delete(regexAndFileNames.urlRegex), `Expected to delete break-on-load information associated with regexp: ${regexAndFileNames.urlRegex}`);
+                    } else {
+                        logger.log(`Stop on entry breakpoint hit but still has remaining files. Keeping: ${bp} that was hit for: ${normalizedMappedUrl} because it's still needed for: ${Array.from(regexAndFileNames.fileSet.entries()).join(", ")}`);

```

Comment:
This means that we will still see the flickering if this particular file/url is reloaded right(since we are not removing the breakpoint)? I guess there is no way to remove a particular match of a setBreakpointByUrl request?

Reply:
It's technically possible to do it if we play with the regexp, but I don't think it's worth the risk or effort to do it.

Reply:
Hmm all right.